In [10]:
import numpy as np
import pandas as pd
import os


"""
This will load a csv file and return a pandas dataframe
@input: filepath:: string :: filepath of csv file
@input: names:: list :: column names list 
"""
def load_csv(filepath, names=None):
    filepath =  os.path.abspath(filepath)
    if os.path.exists(filepath):
        df = pd.read_csv(filepath, names=names, encoding='utf-8')
        return df
    
    return None

filepath = "data/titanic.csv"
# Create a pandas dataframe using file csv data
df = load_csv(filepath)
# shows the first 5 rows in the dataframe
df.head()
    

Survived  Pclass                                               Name  \
0         0       3                             Mr. Owen Harris Braund   
1         1       1  Mrs. John Bradley (Florence Briggs Thayer) Cum...   
2         1       3                              Miss. Laina Heikkinen   
3         1       1        Mrs. Jacques Heath (Lily May Peel) Futrelle   
4         0       3                            Mr. William Henry Allen   

      Sex   Age  Siblings/Spouses Aboard  Parents/Children Aboard     Fare  
0    male  22.0                        1                        0   7.2500  
1  female  38.0                        1                        0  71.2833  
2  female  26.0                        0                        0   7.9250  
3  female  35.0                        1                        0  53.1000  
4    male  35.0                        0                        0   8.0500

In [13]:
# this will give you the data distribution per column
df.describe()

Survived      Pclass         Age  Siblings/Spouses Aboard  \
count  887.000000  887.000000  887.000000               887.000000   
mean     0.385569    2.305524   29.471443                 0.525366   
std      0.487004    0.836662   14.121908                 1.104669   
min      0.000000    1.000000    0.420000                 0.000000   
25%      0.000000    2.000000   20.250000                 0.000000   
50%      0.000000    3.000000   28.000000                 0.000000   
75%      1.000000    3.000000   38.000000                 1.000000   
max      1.000000    3.000000   80.000000                 8.000000   

       Parents/Children Aboard       Fare  
count               887.000000  887.00000  
mean                  0.383315   32.30542  
std                   0.807466   49.78204  
min                   0.000000    0.00000  
25%                   0.000000    7.92500  
50%                   0.000000   14.45420  
75%                   0.000000   31.13750  
max                   6.000000  512.32920

Observation on the data on a high level
1. Total number of passengers on board is 887, of which 38% of people have survived.
2. The average age is 29.4 years with range of 0.42years to 80years.
3. The Average fare is 32.3, with range of 0 to 512.3
4. 75% of population have age 38years or below
5. Atleast 50% of population have purchased 3rd class tickets

In [19]:
df[df["Survived"]==1].count()/df.count()

Survived                   0.385569
Pclass                     0.385569
Name                       0.385569
Sex                        0.385569
Age                        0.385569
Siblings/Spouses Aboard    0.385569
Parents/Children Aboard    0.385569
Fare                       0.385569
dtype: float64

In [23]:
df[df["Pclass"]==3].count()/df.count()

Survived                   0.549042
Pclass                     0.549042
Name                       0.549042
Sex                        0.549042
Age                        0.549042
Siblings/Spouses Aboard    0.549042
Parents/Children Aboard    0.549042
Fare                       0.549042
dtype: float64

In [27]:
df1 = df.groupby(['Sex'])
df1.describe()

Survived                                              Pclass            \
          count      mean       std  min  25%  50%  75%  max  count      mean   
Sex                                                                             
female    314.0  0.742038  0.438211  0.0  0.0  1.0  1.0  1.0  314.0  2.159236   
male      573.0  0.190227  0.392823  0.0  0.0  0.0  0.0  1.0  573.0  2.385689   

        ... Parents/Children Aboard        Fare                              \
        ...                     75%  max  count       mean        std   min   
Sex     ...                                                                   
female  ...                     1.0  6.0  314.0  44.479818  57.997698  6.75   
male    ...                     0.0  5.0  573.0  25.633935  43.267730  0.00   

                                            
              25%      50%   75%       max  
Sex                                         
female  12.071875  23.0000  55.0  512.3292  
male     7.895800  11.1333  27.0  512.3292  

[2 rows x 48 columns]

Observations
1. There were 314 women and 573 men onbaord the titanic which a sex ratio of 1:1.82 women against men.
2. The ratio of women who have survived is 74% against the ratio of men who have survived is 19%